test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2107,7 +2110,53 @@ protected static IndexVersion minimumIndexVersion() throws IOException {
         return minVersion;
     }
 
-    private static Optional<Version> parseLegacyVersion(String version) {
+    /**
+     * Returns the minimum transport version among all nodes of the cluster
+     */
+    protected static TransportVersion minimumTransportVersion() throws IOException {
+        Response response = client.performRequest(new Request("GET", "_nodes"));
+        ObjectPath objectPath = ObjectPath.createFromResponse(response);
+        Map<String, Object> nodesAsMap = objectPath.evaluate("nodes");
+
+        TransportVersion minTransportVersion = null;
+        for (String id : nodesAsMap.keySet()) {
+
+            var transportVersion = getTransportVersionWithFallback(
+                objectPath.evaluate("nodes." + id + ".version"),
+                objectPath.evaluate("nodes." + id + ".transport_version"),
+                () -> TransportVersions.MINIMUM_COMPATIBLE
+            );
+            if (minTransportVersion == null || minTransportVersion.after(transportVersion)) {
+                minTransportVersion = transportVersion;
+            }
+        }
+
+        assertNotNull(minTransportVersion);
+        return minTransportVersion;
+    }
+
+    protected static TransportVersion getTransportVersionWithFallback(
+        Object versionField,
+        Object transportVersionField,
+        Supplier<TransportVersion> fallbackSupplier
+    ) {
+        if (transportVersionField instanceof Integer transportVersionId) {
```

Comment:
This should be checking against `Number`, json can be funny with number types sometimes